hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/BulkInsertDataSetWriter.java

```diff
@@ -130,7 +132,20 @@ public TxnView getTxn(){
         List<Long> allCongloms = Lists.newArrayList();
         allCongloms.add(td.getHeapConglomerateId());
         ArrayList<DDLMessage.TentativeIndex> tentativeIndexList = new ArrayList();
-        
+        DDLDriver ddlDriver=DDLDriver.driver();
+        for(DDLMessage.DDLChange ddlChange : ddlDriver.ddlWatcher().getTentativeDDLs()){
+            DDLMessage.TentativeIndex tentativeIndex = ddlChange.getTentativeIndex();
+            if (tentativeIndex != null) {
+                long table = tentativeIndex.getTable().getConglomerate();
+                TxnView indexTxn = DDLUtils.getLazyTransaction(ddlChange.getTxnId());
+                if (table == td.getHeapConglomerateId() && indexTxn.getCommitTimestamp() > 0 &&
+                    indexTxn.getCommitTimestamp() < txn.getTxnId()) {
```

Comment:
I think you should use a DDLFilter here, it encapsulates the transactional logic in a better way.

Reply:
Daniel and I had a discussion about DDLFilter. We suspect there are issues with it and it does not work for index creation. Changing it may have an impact on dictionary writes. I will file a Jira to investigate DDLFilter issue. 

For bulk index load, I will keep the code as it is for now.